src/App.jsx

```diff
@@ -14,45 +16,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPH = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedList(sortField, isReversed) {
+  const preparedList = [...goodsFromServer];
+
+  if (sortField) {
+    preparedList.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPH:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? preparedList.reverse() : preparedList;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedList = getPreparedList(sortField, isReversed);
```

Comment:
Calling it directly within a component render might not be the most performant option. Use memoization (useMemo) to avoid recalculating the sorted list every render.

Reply:
thanks for your advise, now I know what is useMemo)